packages/itwinui-css/src/side-navigation/side-navigation.scss

```diff
@@ -111,28 +111,28 @@ $iui-side-navigation-icon-margins: calc(1.5 * var(--iui-size-m));
     &,
     &:hover,
     &[disabled] {
-      border-color: var(--iui-color-background-border);
+      border-color: var(--iui-color-border);
```

Comment:
I would be ok with increasing `border-right` from 1px to 2px to align with header's `border-bottom`.  What do you think about removing `border-top` & `border-bottom` from all buttons _except_ for the expander?
![Screenshot 2022-11-03 at 11 47 46 AM](https://user-images.githubusercontent.com/849817/199768772-158069c3-a3a5-4e1d-9ed2-b8db3c63ca13.png) ![Screenshot 2022-11-03 at 11 47 59 AM](https://user-images.githubusercontent.com/849817/199768774-2f9f4d5a-57d3-4cea-ac09-e03f4a6482e7.png)

Reply:
We would need to update the dividing line thickness in the header to match.
![Screenshot 2022-11-03 at 11 52 22 AM](https://user-images.githubusercontent.com/849817/199769411-a5b06678-3799-4cda-9398-f0d8b025e6f9.png)

Reply:
> What do you think about removing `border-top` & `border-bottom` from all buttons _except_ for the expander?

I'm not too familiar with this PR since I'm not a reviewer, but just wanted to mention that I think the borderless buttons add a more modern feel imo.

Reply:
I've increased `border-right` thickness to 2px to align with the header's `border-bottom`.  Everything else we'll revisit at a later date.